### PR TITLE
API pages cleanup

### PR DESCRIPTION
**Pull request by kbatuigas**
# Description
- Ensure consistent usage of CIRCLE_TOKEN env var
- Ensure consistent usage of double quotes with env vars in curl commands as well as request format
- General content refresh

# Reasons
[Jira ticket](https://circleci.atlassian.net/browse/DOCTEAM-703)

# Content Checklist
Please follow our style when contributing to CircleCI docs. Our style guide is here: [https://circleci.com/docs/style/style-guide-overview](https://circleci.com/docs/style/style-guide-overview).

Please take a moment to check through the following items when submitting your PR (this is just a guide so will not be relevant for all PRs) 😸:

- [x] Break up walls of text by adding paragraph breaks.
- [x] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [x] Keep the title between 20 and 70 characters.
- [x] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [x] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [x] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [x] Include relevant backlinks to other CircleCI docs/pages.

### CODE REVIEW

File: jekyll/_cci2/api-developers-guide.md [hunk 1]
```diff
@@ -255,7 +255,7 @@ The following section details the steps you would need, from start to finish, to
         type: string
     ```
 
-    You will need to declare the parameters you expect to receive from the API. In this case, under the `parameters` key, we definte an "image-tag" to be expected in the JSON payload of a POST request to the _Trigger New Pipeline_ endpoint.
+    You will need to declare the parameters you expect to receive from the API. In this case, under the `parameters` key, we define an "image-tag" to be expected in the JSON payload of a POST request to the _Trigger New Pipeline_ endpoint.
```
Comment (devinmarieb) [inline jekyll/_cci2/api-developers-guide.md]:
```suggestion
    You will need to declare the parameters you expect to receive from the API. In this case, under the `parameters` key, we define an `image-tag` to be expected in the JSON payload of a POST request to the [Trigger a new pipeline](https://circleci.com/docs/api/v2/index.html#operation/triggerPipeline) endpoint.
```

File: jekyll/_cci2/api-developers-guide.md [hunk 2]
```diff
@@ -593,23 +596,23 @@ Notice that in this JSON response, you will receive detailed metrics for the set
 - `total credits used` - The total number of credits that were used during the build.
 - `windows_start & windows_end` - The time the build was initiated, and then completed.
 
-**Note** The above example only shows just a few builds. When you run this command, you may receive up to 250 individual builds that you can review in much more detail.
+**Note** The above example only shows a few builds. When you run this command, you may receive up to 250 individual builds that you can review in much more detail.
```
Comment (devinmarieb) [inline jekyll/_cci2/api-developers-guide.md:571]: Should we change this bold not to an official banner note?

File: jekyll/_cci2/api-developers-guide.md [hunk 3]
```diff
@@ -66,26 +65,26 @@ It is recommended that you specify an Accept header in your API requests. The ma
...
+    --header 'Accept: application/json'    \
+    https://circleci.com/api/v2/project/{project-slug}/pipeline
+  ```
 
 ## Getting started with the API
 {: #getting-started-with-the-api }
 
-The CircleCI API shares similarities with previous API versions in that it identifies your projects using repository name. For instance, if you want to pull information from CircleCI about the GitHub repository https://github.com/CircleCI-Public/circleci-cli you can refer to that in the CircleCI API as `gh/CircleCI-Public/circleci-cli`, which is a "triplet" of the project type (VCS provider), the name of your "organization" (or your username), and the name of the repository.
+The CircleCI API shares similarities with previous API versions in that it identifies your projects using repository name. For instance, if you want to pull information from CircleCI about the GitHub repository "https://github.com/CircleCI-Public/circleci-cli" you can refer to that in the CircleCI API as `gh/CircleCI-Public/circleci-cli`, which is a "triplet" of the project type (VCS provider), the name of your "organization" (or your username), and the name of the repository.
```
Comment (devinmarieb) [inline jekyll/_cci2/api-developers-guide.md:87]: We don't use quotes very often and I'm unsure if they fit here either. Maybe change "organization" to be more specific about the role and put it in bold like we do elsewhere when we reference things about roles? I'm also not sure the link belongs in a quote. Will the link still be clickable in quotes? If so, I think the quotes will make it seem more crowded. If the quotes escape the link and it's not clickable, then I do think they make sense in that instance.

File: jekyll/_cci2/api-developers-guide.md [hunk 4]
```diff
@@ -227,29 +229,31 @@ The following section details the steps you would need, from start to finish, to
...
-          - run: echo "Image tag used was ${IMAGETAG}"
-    parameters:
-      image-tag:
-        default: latest
-        type: string
-    ```
-
-    You will need to declare the parameters you expect to receive from the API. In this case, under the `parameters` key, we define an "image-tag" to be expected in the JSON payload of a POST request to the _Trigger New Pipeline_ endpoint.
-
+6. Let's move on to a more complex example: triggering a pipeline and passing a parameter that can be dynamically substituted into your configuration. In this example, we will pass a Docker image tag to our `docker` executor key.
```
Comment (devinmarieb) [inline jekyll/_cci2/api-developers-guide.md:232]:
```suggestion
6. Let us move on to a more complex example: triggering a pipeline and passing a parameter that can be dynamically substituted into your configuration. In this example, we will pass a Docker image tag to our `docker` executor key. 
```

File: jekyll/_cci2/api-developers-guide.md [hunk 5]
```diff
@@ -227,29 +229,31 @@ The following section details the steps you would need, from start to finish, to
...
+            IMAGETAG: "<< pipeline.parameters.image-tag >>"
+          steps:
+            - run: echo "Image tag used was ${IMAGETAG}"
+      parameters:
+        image-tag:
+          default: latest
+          type: string
+      ```
+
+      You will need to declare the parameters you expect to receive from the API. In this case, under the `parameters` key, we define an "image-tag" to be expected in the JSON payload of a POST request to the _Trigger New Pipeline_ endpoint.
```
Comment (devinmarieb) [inline jekyll/_cci2/api-developers-guide.md:255]:
```suggestion    
      You will need to declare the parameters you expect to receive from the API. In this case, under the `parameters` key, we define an `image-tag` to be expected in the JSON payload of a POST request to the [Trigger a new pipeline](https://circleci.com/docs/api/v2/index.html#operation/triggerPipeline) endpoint.
```

File: jekyll/_cci2/api-developers-guide.md [hunk 6]
```diff
@@ -268,24 +272,16 @@ endpoints currently available.
 ## Additional API use cases
 {: #additional-api-use-cases }
 
-Now that you have a general understanding of how the CircleCI API v2 service works through an end-to-end API example request and walkthrough, let's look at a few common tasks and operations you may perform on a regular basis when using the API. Whether you wish to return information about a job or project, or retrieve more detailed information about a project by reviewing its artifacts, the examples shown below should assist you in gaining a better understanding of how to make some API requests to the server so you can perform a deep dive into the specifics of your work.
+Now that you have a general understanding of how the CircleCI API v2 service works through an end-to-end API example request and walkthrough, let's look at a few common tasks and operations you may perform on a regular basis when using the API.
```
Comment (devinmarieb) [inline jekyll/_cci2/api-developers-guide.md:275]:
```suggestion
Now that you have a general understanding of how the CircleCI API v2 service works through an end-to-end API example request and walkthrough, let us look at a few common tasks and operations you may perform on a regular basis when using the API.
```

File: jekyll/_cci2/api-developers-guide.md [hunk 7]
```diff
@@ -268,24 +272,16 @@ endpoints currently available.
...
-
-```
-{project_type}/{org_name}/{repo_name}
-```
-
-The `project_slug` is included in the payload when you pull information about a project, which enables you to retrieve detailed information about a specific project.
+You may often find it helpful to retrieve information about a specific project, including the name of the organization the project belongs to, the version control system (VCS) that hosts the project, and other details. The CircleCI API enables you to return this and other information by making a single GET request to the `project/{project-slug}` endpoint by passing the `project-slug` parameter.
 
-**Note** If you would like more detailed information about a project, or simply need a refresher on the specifics of a project, please refer to the CircleCI [Projects]({{site.baseurl}}/projects/) page.
+**Note** If you would like more detailed information about a project, or need a refresher on the specifics of a project, please refer to the CircleCI [Projects]({{site.baseurl}}/projects/) page.
```
Comment (devinmarieb) [inline jekyll/_cci2/api-developers-guide.md:284]: Should this remain a bold note or be turned into an official note banner?

File: jekyll/_cci2/api-developers-guide.md [hunk 8]
```diff
@@ -327,9 +323,9 @@ Notice in the example above that you will receive very specific information abou
 ### Get job details
 {: #get-job-details }
 
-Much like the Get Project Details API request described in the previous example, the Get Job Details API request enables you to return specific job information from the CircleCI API by making a single API request. Retrieving job information can be very useful when you want information about how your job performed, what resources were used (e.g. pipeline, executor type, etc.), and the time it took for the job to finish.
+Much like the Get Project Details API request described in the previous example, the Get Job Details API request enables you to return specific job information from the CircleCI API by making a single API request.
```
Comment (devinmarieb) [inline jekyll/_cci2/api-developers-guide.md:326]: The "Get project details" endpoint was either mistyped or has changed -- it's now called "Get a project".
```suggestion
Much like the Get a project API request described in the previous example, the [Get job details](https://circleci.com/docs/api/v2/index.html#operation/getJobDetails) API request enables you to return specific job information from the CircleCI API by making a single API request. 
```

**Review by devinmarieb — APPROVED** [review]

File: jekyll/_cci2/api-developers-guide.md [hunk 9]
```diff
@@ -350,17 +324,18 @@ Notice in the example above that you will receive very specific information abou
...
+Much like the Get a project API request described in the previous example, the [Get job details](https://circleci.com/docs/api/v2/index.html#operation/getJobDetails) API request enables you to return specific job information from the CircleCI API by making a single API request. 
 
-Please remember, jobs are collections of steps. Each job must declare an executor that is either `docker`, `machine`, `windows` or `macos`. `machine` includes a default image if not specified, for `docker` you must specify an image to use for the primary container, for `macos` you must specify an Xcode version, and for `windows` you must use the Windows orb.
+Retrieving job information can be very useful when you want information about how your job performed, what resources were used (e.g. pipeline, executor type, etc.), and the time it took for the job to finish.
 
 #### Steps
 {: #steps }
 {:.no_toc}
```
Comment (rosieyohannan) [inline jekyll/_cci2/api-developers-guide.md:333]: Could you take out the no_tocs so all sections appear in the right hand nav?

**Pull request merged**